app/controllers/api/base_controller.rb

```diff
@@ -187,7 +187,11 @@ def authorize
         return false
       end
 
-      unless authenticate
+      if authenticate
+        if User.current.disabled
+          render_error('custom_error', :status => :unauthorized, locals: {message: _('User is disabled')})
+        end
```

Comment:
This should probably return false. Also should we logout user? I haven't tested myself, but feels the `authenticate` method would create valid user session. Which would probably be rejected on next page load again.

Reply:
This is only here to insure, that disabled user can't abuse he is already logged in, right? Marek's note would be only valid for `sso` authentications. Could we maybe do something about those?